src/Agent.Listener/Configuration/ConfigurationProvider.cs

```diff
@@ -208,6 +223,45 @@ public void UpdateAgentSetting(AgentSettings settings)
             settings.ProjectName = _projectName;
         }
 
+        public bool GetTagsRequired(CommandSettings command)
+        {
+            return command.GetMachineGroupTagsRequired();
+        }
+
+        public async Task GetAndAddTags(CommandSettings command, TaskAgent agent)
+        {
+            string tagString = command.GetMachineGroupTags();
+            Trace.Info("Given tags - {0} will be processed and added", tagString);
+
+            if (!string.IsNullOrWhiteSpace(tagString))
+            {
+                var tags =
```

Comment:
"splitoption.removeempty" not able to remove the white space entries. I tried this but it did not work. CanEnsureMachineGroupAgentConfigureVSTSScenarioWithTags() L0 test fails with it